core-java-modules/core-java-arrays-operations-basic-2/src/test/java/com/baeldung/array/allelementsareequal/AllElementsAreEqualInArrayUnitTest.java

```diff
@@ -0,0 +1,144 @@
+    private final static int[] INT_ARRAY_SINGLE_EL = { 42 };
+
+    public <T> boolean isAllEqual(T[] array) {
+        if (array == null || array.length == 0) {
+            return false;
+        }
+        for (int i = 1; i < array.length; i++) {
+            if (!Objects.equals(array[0], array[i])) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Test
+    void whenUsingArraysAreEqual_thenCorrect() {
+        assertTrue(isAllEqual(ARRAY_ALL_EQ));
+        assertTrue(isAllEqual(ARRAY_ALL_NULL));
+        assertTrue(isAllEqual(ARRAY_SINGLE_EL));
+
+        assertFalse(isAllEqual(ARRAY_NOT_EQ));
+        assertFalse(isAllEqual(ARRAY_EMPTY));
+        assertFalse(isAllEqual(ARRAY_NULL));
+    }
+
+    public <T> boolean isAllEqualByDistinct(T[] array) {
+        if (array == null || array.length == 0) {
+            return false;
+        }
+        return Arrays.stream(array)
+            .distinct()
+            .count() == 1;
+    }
+
+    @Test
+    void whenUsingArraysAreEqualByDistinct_thenCorrect() {
+        assertTrue(isAllEqualByDistinct(ARRAY_ALL_EQ));
+        assertTrue(isAllEqualByDistinct(ARRAY_ALL_NULL));
+        assertTrue(isAllEqualByDistinct(ARRAY_SINGLE_EL));
+
+        assertFalse(isAllEqualByDistinct(ARRAY_NOT_EQ));
+        assertFalse(isAllEqualByDistinct(ARRAY_EMPTY));
+        assertFalse(isAllEqualByDistinct(ARRAY_NULL));
+    }
+
+    public <T> boolean isAllEqualByAllMatch(T[] array) {
+        if (array == null || array.length == 0) {
+            return false;
+        }
+        return Arrays.stream(array)
+            .allMatch(element -> Objects.equals(array[0], element));
+    }
+
+    @Test
+    void whenUsingArraysAreEqualByAllMatch_thenCorrect() {
+        assertTrue(isAllEqualByAllMatch(ARRAY_ALL_EQ));
+        assertTrue(isAllEqualByAllMatch(ARRAY_ALL_NULL));
+        assertTrue(isAllEqualByAllMatch(ARRAY_SINGLE_EL));
+
+        assertFalse(isAllEqualByAllMatch(ARRAY_NOT_EQ));
+        assertFalse(isAllEqualByAllMatch(ARRAY_EMPTY));
+        assertFalse(isAllEqualByAllMatch(ARRAY_NULL));
+    }
+
+    // primitive array approaches below (int[] as example)
+
+    public boolean isAllEqual(int[] array) {
+        if (array == null || array.length == 0) {
+            return false;
+        }
+        for (int i = 1; i < array.length; i++) {
+            if (!Objects.equals(array[0], array[i])) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Test
+    void whenUsingArraysAreEqualForIntArrays_thenCorrect() {
+        assertTrue(isAllEqual(INT_ARRAY_ALL_EQ));
+        assertTrue(isAllEqual(INT_ARRAY_SINGLE_EL));
+
+        assertFalse(isAllEqual(INT_ARRAY_NOT_EQ));
+    }
+
+    public boolean isAllEqualByDistinct(int[] array) {
+        if (array == null || array.length == 0) {
+            return false;
+        }
+        return Arrays.stream(array)
```

Comment:
We don't need to do the array.length == 0 check, it's already handled by Arrays.stream (same elsewhere)
```suggestion
        if (array == null) {
            return false;
        }
        return Arrays.stream(array)
```

Reply:
That's true. But I would leave the length == 0 check as a pre-check. If it's empty, we can immediately return the result without creating unnecessary objects, such as Stream, ArraySpliterator, etc. what do you think?